File: hw/pilatasig/u2/ZooApp JavaScript/ec/edu/espe/zoo/view/ZooApp.js
// Importar las clases desde sus respectivos archivos
import { Animal } from "../model/Animal.js";
import { Cage } from "../model/Cage.js";
import { Mammal } from "../model/Mammal.js";
import { Vaccine } from "../model/Vaccine.js";

console.log("Hello World!");
console.log("This is David's Zoo");

let animal;
let animalId;
let name;
let bornOnDate;
let weight;
let sex;
let vaccines = [];
let animals = [];

// Definir los datos del ejemplo
let vaccineId = 1;
let vaccineName = "Paracetamol";

animalId = 1;
name = "Lucy";
bornOnDate = new Date(2024, 0, 21); // Meses en JavaScript van de 0 a 11
weight = 4.57;
sex = "F";

// Crear una vacuna
let vaccine = new Vaccine(vaccineId, vaccineName);
vaccines.push(vaccine);

// Crear un animal
animal = new Animal(animalId, name, bornOnDate, weight, sex, vaccines);
console.log("My animal is --> \n" + animal.toString());

// Crear un mamífero
let mammal = new Mammal(0.21, 2, animalId, name, bornOnDate, weight, sex, vaccines);
console.log("My mammal is --> " + mammal.toString());

// Crear una jaula
let cage = new Cage(1, animals);
mammal.assignCage(cage);

// Añadir una nueva vacuna
vaccine = new Vaccine(5, "Ibuprofen");
vaccines.push(vaccine);

// Métodos adicionales del mamífero
mammal.breastFeed();
mammal.register();
mammal.provideVaccination(vaccine);

console.log("Cage --> " + cage.toString());

File: ws/pilatasig/u2/ws15Inheritances/ZooApp/src/main/java/ec/edu/espe/zooapp/view/ZooApp.java
package ec.edu.espe.zooapp.view;

import ec.edu.espe.zooapp.model.Animal;
import ec.edu.espe.zooapp.model.Cage;
import ec.edu.espe.zooapp.model.Mammal;
import ec.edu.espe.zooapp.model.Vaccine;
import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author David Pilatasig
 */
public class ZooApp {

    public static void main(String[] args) {
        System.out.println("Hello World!");
        System.out.println("This is David's Zoo");
        
        Animal animal;
        int animalId;
        String name;
        Date bornOnDate;
        float weight;
        char sex;
        ArrayList<Vaccine> vaccines;
        ArrayList<Animal> animals;
        animals= new ArrayList<>();
        
        int vaccineId = 1;
        String vaccineName;
        vaccineName="Paracetamol";
        
        animalId=1;
        name="Lucy";
        bornOnDate=new Date(2024, 0, 21);
        weight= 4.57F;
        sex='F';
       
        Vaccine vaccine;
        
        vaccine= new Vaccine(vaccineId, vaccineName);
        vaccines= new ArrayList<>();
        vaccines.add(vaccine);
        
        animal= new Animal(animalId, name, bornOnDate, weight, sex, vaccines);
        
        
        System.out.println("My animal is --> \n" + animal);
        
        Mammal mammal= new Mammal(0.21F, 2, animalId, name, bornOnDate, weight, sex, vaccines);
        System.out.println("My mammal is --> " + mammal);
        
        
        Cage cage;
        cage= new Cage(1, animals);
        
        
        mammal.assignCage(cage);
        
        
        
        vaccine=new Vaccine(5,"Ibuprofhene");
        vaccines.add(vaccine);
        
        
        mammal.breastFeed();
        mammal.register();
        mammal.provideVaccination(vaccine);
        
        System.out.println("Cage -->" + cage);
    }
}
